report/node.go

```diff
@@ -118,28 +117,16 @@ func (n Node) WithAdjacent(a ...string) Node {
 	return n
 }
 
-// WithLatestActiveControls returns a fresh copy of n, with active controls cs added to LatestControls.
+// WithLatestActiveControls says which controls are active on this node.
+// Implemented as a delimiter-separated string in Latest
 func (n Node) WithLatestActiveControls(cs ...string) Node {
-	lcs := map[string]NodeControlData{}
-	for _, control := range cs {
-		lcs[control] = NodeControlData{}
-	}
-	return n.WithLatestControls(lcs)
-}
-
-// WithLatestControls returns a fresh copy of n, with lcs added to LatestControls.
-func (n Node) WithLatestControls(lcs map[string]NodeControlData) Node {
-	ts := mtime.Now()
-	for k, v := range lcs {
-		n.LatestControls = n.LatestControls.Set(k, ts, v)
-	}
-	return n
+	return n.WithLatest(NodeActiveControls, mtime.Now(), strings.Join(cs, ScopeDelim))
 }
 
-// WithLatestControl produces a new Node with control added to it
-func (n Node) WithLatestControl(control string, ts time.Time, data NodeControlData) Node {
-	n.LatestControls = n.LatestControls.Set(control, ts, data)
-	return n
+// ActiveControls returns a string slice with the names of active controls.
+func (n Node) ActiveControls(cs ...string) []string {
```

Comment:
`cs` doesn't seem to be used in the function or am I missing something? Weird how linter didn't complain:

```suggestion
func (n Node) ActiveControls() []string {
```

Reply:
Right, it was bad copy/paste from the method above, thanks.
Go doesn't complain about unused parameters inside the function because they may be needed to conform to an interface.  And it's ok to supply an empty list when calling a `...` function.